utils/https.h

```diff
@@ -34,17 +52,26 @@ void https_logger_init();
 int https_logger_release();
 
 /**
- * @brief Send message via HTTP(S) protocol
+ * @brief Send a POST request message via HTTP(S) protocol
  *
- * @param[in] host HTTP(S) host
- * @param[in] port HTTP(S) port
- * @param[in] api API path for POST request to HTTP(S) server, i.e "transaction/". It must be in string.
- * @param[in] msg Message to send
- * @param[in] ssl_seed Seed for ssl connection
+ * @param[in, out] ctx The pointer points to http context
  *
  * @return #status_t
  */
-status_t send_https_msg(char const *host, char const *port, char const *api, const char *msg, const char *ssl_seed);
+status_t send_https_msg(https_ctx_t* ctx);
+
+/**
+ * @brief Callback function for http parser
+ *
+ * @param[in] parser HTTP(S) parser
+ * @param[in] at HTTP(S) message to parse
+ * @param[in] length Length of text at
+ *
+ * @return
+ * - 0 on success
+ * - non-zero on error
+ */
+int https_parser_callback(http_parser* parser, const char* at, size_t length);
```

Comment:
This function can be declared as an internal one and initialized via customized initialization process. Reducing the exposed symbols can aid compiler optimizations.